File: src/main/java/com/github/joelws/release/tracker/service/release/operation/DeleteReleaseServiceOperation.java
package com.github.joelws.release.tracker.service.release.operation;

import com.github.joelws.release.tracker.service.ServiceHelper;
import com.github.joelws.release.tracker.service.ServiceOperation;
import com.github.joelws.release.tracker.service.release.execution.DeleteReleaseServiceExecution;

import javax.ws.rs.core.Response;

public class DeleteReleaseServiceOperation extends ServiceOperation<String> {
    private final ServiceHelper helper;
    private final DeleteReleaseServiceExecution deleteReleaseServiceExecution;

    public DeleteReleaseServiceOperation(final ServiceHelper helper,
                                         final DeleteReleaseServiceExecution deleteReleaseServiceExecution) {
        this.helper = helper;
        this.deleteReleaseServiceExecution = deleteReleaseServiceExecution;
    }

    @Override
    protected Response delegate(String identifier) {

        deleteReleaseServiceExecution
                .execute(identifier);

        return helper.getRestResponseBuilder().build(200);
    }
}

File: src/main/kotlin/com/github/joelws/release/tracker/service/release/operation/DeleteReleaseServiceOperation.kt
package com.github.joelws.release.tracker.service.release.operation

import com.github.joelws.release.tracker.response.RestResponse.Success
import com.github.joelws.release.tracker.response.build
import com.github.joelws.release.tracker.service.ServiceExecution
import com.github.joelws.release.tracker.service.ServiceHelper
import com.github.joelws.release.tracker.service.ServiceOperation
import javax.ws.rs.core.Response

open class DeleteReleaseServiceOperation(val helper: ServiceHelper,
                                         val deleteReleaseServiceExecution: ServiceExecution<String?, Unit>) : ServiceOperation<String> {

    override fun delegate(param: String?): Response {

        deleteReleaseServiceExecution
                .execute(param)

        return Success().build()
    }
}
